### PR TITLE
#232: require both input file and opendap url unless the user only wa…

### LINKED ISSUE #232: The MMD fields "data access" and "storage information->file location" are the same when the netcdf product is an opendap url
Storage info should be a file and not an opendap url. Currently these are the same, and equal to `netcdf_product`. 

Todo:

- [ ] We need to change the input to `Nc_to_mmd.__init__` from `netcdf_product` to `netcdf_file` and `opendap_url`. The netcdf files can still be tested locally if we use it with `check_only=True`.

### PR DESCRIPTION
**Pull request by mortenwh**
…nt to check the nc file with check_only=True

**Summary**: see #232 

**Related issue**: closes #232 

**Suggested reviewer(s)**:

**Reviewer checklist**:

- [ ] The headers of all files contain a reference to the repository license (i.e., "License: This file is part of py-mmd-tools, licensed under the Apache License 2.0 (https://www.apache.org/licenses/LICENSE-2.0)")
- [ ] 100% test coverage of new code - meaning:
    - [ ] The overall test coverage increased or remained the same as before
    - [ ] Every function is accompanied with a test suite
    - [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
    - [ ] Functions with optional arguments have separate tests for all options
- [ ] Examples are supported by doctests
- [ ] All tests are passing
- [ ] All names (e.g., files, classes, functions, variables) are explicit
- [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

File: py_mmd_tools/nc_to_mmd.py [hunk 1]
```diff
@@ -1264,49 +1269,34 @@ def to_mmd(self, collection=None, checksum_calculation=False, mmd_yaml=None,
...
 
         # Set storage_information
-
-        file_location = self.netcdf_product
-
         self.metadata['storage_information'] = {
-            'file_name': os.path.basename(self.netcdf_product),
-            'file_location': file_location,
+            'file_name': os.path.basename(self.netcdf_file),
+            'file_location': os.path.dirname(self.netcdf_file),
```
Comment (johtoblan) [inline py_mmd_tools/nc_to_mmd.py:1286]: Only works if full path is given.
```
>>> os.path.dirname("hello.txt")
''
>>> os.path.dirname("~/hello.txt")
'~'
>>> os.path.dirname("home/user/hello.txt")
'home/user'
```

File: py_mmd_tools/script/nc2mmd.py [hunk 2]
```diff
@@ -56,23 +66,35 @@ def create_parser():
 
 def main(args=None):
     """Run tool to create MMD xml file from input netCDF-CF file(s)"""
+    if not args.dry_run:
+        if args.url is None:
+            raise ValueError('OPeNDAP url must be provided')
+        if args.output_dir is None:
+            raise ValueError('MMD XML output directory must be provided')
 
+    assume_base_url = False
```
Comment (johtoblan) [inline py_mmd_tools/script/nc2mmd.py:75]: Hva gjør denne? Hvis vi gir en folder som input så antar vi at opendap-linkene har samme rot?

Comment (mortenwh) [reply py_mmd_tools/script/nc2mmd.py:75]: `base_url` henviser til filnavnet - some `os.path.basename(path_to_file)` henviser til filnavn i `path_to_file`

File: py_mmd_tools/nc_to_mmd.py (hunk 1 above)
Comment (mortenwh) [reply py_mmd_tools/nc_to_mmd.py:1286]: hmm - må vi legge inn mer da?

Comment (mortenwh) [reply py_mmd_tools/nc_to_mmd.py:1286]: kunne sjekket at det er en faktisk sti som begynner med "/" og ikke inneheolder "~". Eller kanskje det finnes en funksjon som kan brukes allerede?

Comment (mortenwh) [reply py_mmd_tools/nc_to_mmd.py:1286]: uansett, så er det noe ansvar som må legges på MMD-produsenten..

Comment (mortenwh) [reply py_mmd_tools/nc_to_mmd.py:1286]: ja, vi bør kreve full path

File: py_mmd_tools/script/nc2mmd.py (hunk 2 above)
Comment (mortenwh) [reply py_mmd_tools/script/nc2mmd.py:75]: kunne kanskje endret til `url_basename`?

Comment (johtoblan) [reply py_mmd_tools/script/nc2mmd.py:75]: ja, eller bare legge til noen kommentarer

File: py_mmd_tools/nc_to_mmd.py (hunk 1 above)
Comment (mortenwh) [reply py_mmd_tools/nc_to_mmd.py:1288]: løst i bb55c5c02fc28cc209a9a954095176bb4da241e0, men må lage test...

**Review by johtoblan — APPROVED** [review]
LGTM

**Pull request merged**